gslib/gcs_json_credentials.py

```diff
@@ -230,6 +239,15 @@ def _CheckAndGetCredentials(logger):
     # realize their credentials are invalid.
     raise
 
+  if _HasImpersonateServiceAccount() and creds:
```

Comment:
Since you're not doing anything with the exception, why not just move this up to line 220 and remove the wrapping try block?

 if _HasImpersonateServiceAccount() and creds:
  return _GetImpersonationCredentials(creds, logger)
else:
  return creds

Reply:
I actually don't want to use the exception handling in 220, but I'll shift it around to use the existing `except:`